services/brig/src/Brig/Code.hs

```diff
@@ -160,13 +161,25 @@ mkGen cfor = liftIO $ do
       let key = mkKey d (Text.encodeUtf8 (emailKeyUniq (mkEmailKey e)))
           val = Value . unsafeRange . Ascii.encodeBase64Url <$> randBytes 15
        in Gen cfor key val
```

Comment:
Maybe for symmetry you can factor this out into a function `genEmailLinkGen` next to `mk6DigitGen'`?

I thought I could restructure this to look easier, but now I think I was wrong, it looks mostly sane :+1:   (except for the convoluted requirements due to our lack of understanding of the code ealier that we don't want to touch any more).